packages/shared/src/lib/core/activity/types/evm/base-evm-activity.type.ts

```diff
@@ -1,10 +1,12 @@
 import { NetworkNamespace } from '@core/network/enums'
 import { BaseActivity } from '../base-activity.type'
 import { ISmartContractSubject } from '@core/wallet'
+import { EvmNetworkId } from '@core/network'
 
 export type BaseEvmActivity = BaseActivity & {
     namespace: NetworkNamespace.Evm
 
+    sourceNetworkId: EvmNetworkId
```

Comment:
What if we will be able to merge L1 -> L2 into one activity? Does it give added value to specify a `sourceNetworkId` at this level?

Reply:
we have `sourceNetworkId: NetworkId` and `destinationNetworkId: NetworkId` in the `BaseActivity` because Stardust activities also have those. 

We know that all Evm activities come from an evm chain, so here we can narrow the type for the source, but not for the destination as L2 -> L1 and L2 -> L2 are possible.

Same goes for Stardust activities. We know that those are are always output based, so the `sourceNetworkId` can be narrrowed to `StardustNetworkId`.